src/containers/AdminContainer/AdminContainer.js

```diff
@@ -1,15 +1,147 @@
+        this.setState({
+            createEmergency: true,
+            updateEmergency: false,
+        });
+    }
+
+    updateEmergency = () => {
+        this.setState({
+            createEmergency: false,
+            updateEmergency: true,
+        });
+    }
+
+    handleInputChange = (event) => {
+        const target = event.target;
+        const name = target.name;
+        this.setState({
+            [name]: value,
+        })
+    }
+
+    handleNewSubmit = (event) => {
+        const emergency = {
+            title: this.state[title],
+            description: this.state[description],
+            contactName: this.state[contactName],
+            contactEmail: this.state[contactEmail],
+            contactPhone: this.state[contactPhone],
+        }
+        createPost(emergency).catch( (error) => {
+            console.log('Error creating post', error);
+        });
+
+        event.preventDefault();
+    }
+
+    handleUpdateSubmit = (event) => {
+        createPost(emergency).catch( (error) => {
+            console.log('Error creating post', error);
+        });
+        event.preventDefault();
+    }
+
+    showNewEmergency = () => {
+        return (
+            <div>
+                <p>Report a new emergency. Make sure all details are correct before publishing.</p>
+                <form onSubmit={this.handleNewSubmit}>
+                    <div class="inputGroup">
+                        <label for="title">Title</label>
+                        <input type="text" name="title" id="title"/>
+                    </div>
+                    <div class="inputGroup">
+                        <label for="description">Description</label>
+                        <input type="text" name="description" id="description"/>
+                    </div>
+                    <div class="inputGroup">
+                        <label for="contactName">Contact Name</label>
+                        <input type="text" name="contactName" id="contactName"/>
+                    </div>
+                    <div class="inputGroup">
+                        <label for="contactEmail">Contact Email</label>
+                        <input type="email" name="contactEmail" id="contactEmail"/>
+                    </div>
+                    <div class="inputGroup">
+                        <label for="contactPhone">Contact Phone</label>
+                        <input type="tel" name="contactPhone" id="contactPhone"/>
+                    </div>
+                </form>
+            </div>
+        );
+    }
+
+    showEmergencies = () => {
+        // TODO : determine how to display all current emergencies, and allow admin
+        // to choose one to update
+        return (
+            <div></div>
+        );
+    }
+
+    showUpdateEmergency = () => {
         return (
             <div>
-                admin stuff
+                <p>Update an existing emergency.</p>
+                <form onSubmit={this.handleUpdateSubmit}>
```

Comment:
any particular reason for a form?

Reply:
Form to me means will have some extra functionality to use when we submit the data whereas we know exactly what we want to do. I'd say just have a button that handle the submit rather than using a form tag. If you have a reason however then please comment below